### PR TITLE
Fixes #35408 - Add time_to_pickup setting

### PR DESCRIPTION
**Pull request by adamruzicka**
overridable by the usual parameter hierarchy and per job.

### CODE REVIEW

**Review by ofedoren — COMMENTED** [review]
Thanks, @adamruzicka, LGTM, but I don't see where this param is used. Is there a corresponding PR for e.g. proxy?

**Comment by adamruzicka** [comment]
> Me, circa yesterday: "I'll open the proxy PR, then the foreman one and link the proxy one there"
> Me, after just opening the PRs: "Another job well done"

Yes, of course, here it is https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/92

File: lib/foreman_remote_execution/engine.rb [hunk 1]
```diff
@@ -155,6 +155,11 @@ class Engine < ::Rails::Engine
               default: 'Job - Invocation Report',
               full_name: N_('Job Invocation Report Template'),
               collection: proc { ForemanRemoteExecution.job_invocation_report_templates_select }
+            setting 'remote_execution_time_to_pickup',
+              type: :integer,
+              description: N_('Time in seconds within which the host has to pick up a job. If the job is not picked up within this limit, the job will be cancelled.'),
```
Comment (ofedoren) [inline lib/foreman_remote_execution/engine.rb:160]: Should we mention that by default it's 24 hours?

**Review by ofedoren — COMMENTED** [review]
Small nit, otherwise LGTM.

File: lib/foreman_remote_execution/engine.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/foreman_remote_execution/engine.rb]: Done

**Review by ofedoren — APPROVED** [review]
Thanks, @adamruzicka, let's get this in.

**Pull request merged**